modules/deployment-model-impl/src/main/java/module-info.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+/**
+ * Mule Deployment Model Implementation Module
+ *
+ * @moduleGraph
+ * @since 4.6
+ */
+module org.mule.runtime.deployment.model.impl {
```

Comment:
beware, there is a split package (`org.mule.runtime.deployment.model.internal.domain`) between deployment model jar and its test jar